lib/api/api-fetch/headers.js

```diff
@@ -94,12 +98,20 @@ class Headers {
   constructor (init = {}) {
     // validateObject allowArray = true
     if (!Array.isArray(init) && typeof init !== 'object') {
-      throw new InvalidArgumentError('The argument \'init\' must be one of type Object or Array')
+      throw new TypeError('The argument \'init\' must be one of type Object or Array')
```

Comment:
@mcollina: In regards to errors? Should we do exactly what the spec says and throw generic `TypeError`s or do we want to enrich them?

Reply:
These changes came from Node core PR: https://github.com/nodejs/node/pull/38986/files?file-filters%5B%5D=.js&file-filters%5B%5D=.mjs#diff-3569a54af4e25669eee4cc77bd0f4e86ca5fafb0cf766759a4bd2fb482fa0984R138

I would lean towards returning TypeErrors directly as that would be better for interop